perses/annihilation/relative.py

```diff
@@ -2635,3 +2640,1611 @@ def _handle_exceptions(self):
+    - is_nonrest_solvent - indicates whether the atom is outside the rest region and is solvent (1), otherwise 0
+    These per particle parameters are defined to allow for scaling of rest-solvent interactions by beta/beta0, if desired. 
+    If the scaling of rest-solvent interactions is desired, the functional form will need to be changed.
+
+    In order to avoid singularities in the alchemical region (specifically in interactions of unique old/new atoms),
+    we introduce a decoupling in the 4th dimensional distance. Decoupling does not affect core or env atoms.
+    The unique old/new atoms are 'lifted' into the 4th dimension 'w' upon 'alchemification' and the length of the 4th dimension is
+    given by r_cutoff * w_scale. w_scale is a user-defined scalar between 0 (non-inclusive) and 1. Taking 'w_scale' to 1
+    means that once the term is maximally 'lifted' into the 4th dimension, the 4th dimension is effectively of length 'r_cutoff'.
+
+    For the first NonbondedForce, global parameters (to be used with particle parameter offsets) are defined to allow
+    for alchemical scaling, but not rest scaling. Computation of contributions from the direct space is disabled.
+
+    For the second NonbondedForce, there is no alchemical or rest scaling. Long range correction is on.
+
+    This class can be tested using perses.tests.utils.validate_endstate_energies_point() and
+    perses.tests.utils.validate_endstate_energies_md(), as is done by perses.tests.test_relative.test_RESTCapableHybridTopologyFactory_energies
+
+    """
+    
+    # Constants copied from: https://github.com/openmm/openmm/blob/master/platforms/reference/include/SimTKOpenMMRealType.h#L89. These will be imported directly once we have addresssed https://github.com/choderalab/openmmtools/issues/522
+    M_PI = 3.14159265358979323846
+    E_CHARGE = (1.602176634e-19)
+    AVOGADRO = (6.02214076e23)
+    EPSILON0 = (1e-6*8.8541878128e-12/(E_CHARGE*E_CHARGE*AVOGADRO))
+    ONE_4PI_EPS0 = (1/(4*M_PI*EPSILON0))
+    #from openmmtools.constants import ONE_4PI_EPS0
+
+    _default_electrostatics_expression_list = [
+
+        "U_electrostatics;",
+
+        # Define electrostatics functional form
+        f"U_electrostatics = {ONE_4PI_EPS0} * chargeProd  * erfc(alpha * r_eff_electrostatics)/ r_eff_electrostatics;",
+
+        # Define chargeProd (with REST scaling)
+        "chargeProd = charge1 * electrostatics_rest_scale1 * charge2 * electrostatics_rest_scale2;",
+
+        # Define alpha
+        "alpha = {alpha_ewald};"
+
+        # Define r_eff
+        "r_eff_electrostatics = sqrt(r^2 + w_electrostatics^2);",
+
+        # Define 4th dimension terms
+        "w_electrostatics = w_scale * r_cutoff * (is_unique_old * lambda_alchemical_electrostatics_new + is_unique_new * lambda_alchemical_electrostatics_old);", # because we want w for unique old atoms to go from 0 to 1 and the opposite for unique new atoms
```

Comment:
@zhang-ivy : I just realized that you might want this to be 
```
        "w_electrostatics = w_scale * r_cutoff * (is_unique_old * (1-lambda_alchemical_electrostatics_old) + is_unique_new * (1-lambda_alchemical_electrostatics_new));", 
```
if if you want `lambda_electrostatics_old` to go from 1 (old atom is fully on) to 0 (old atom is fully off)
```

Reply:
fixed!